geode-core/src/main/java/org/apache/geode/internal/tcp/ConnectionTable.java

```diff
@@ -494,20 +494,28 @@ Connection getThreadOwnedConnection(DistributedMember id, long startTime, long a
 
     // Update the list of connections owned by this thread....
 
-    if (this.threadConnectionMap == null) {
+    ConcurrentMap cm = this.threadConnectionMap;
+    if (cm == null) {
       // This instance is being destroyed; fail the operation
       closeCon(
           "Connection table being destroyed",
           result);
       return null;
     }
 
-    ArrayList al = (ArrayList) this.threadConnectionMap.get(id);
+    ArrayList al = (ArrayList) cm.get(id);
     if (al == null) {
       // First connection for this DistributedMember. Make sure list for this
       // stub is created if it isn't already there.
       al = new ArrayList();
 
+      if (this.threadConnectionMap == null) {
+        // This instance is being destroyed; fail the operation
+        closeCon(
+            "Connection table being destroyed",
+            result);
+        return null;
+      }
       // Since it's a concurrent map, we just try to put it and then
       // return whichever we got.
       Object o = this.threadConnectionMap.putIfAbsent(id, al);
```

Comment:
A simplified version of this looks more like:
```java
...
final Map local = this.member;
if (null == local) {
  return null;
}
final List list = local.computeIfAbsent(id, ArrayList::new);
...
```


Use `JavaWorkarounds.computeIfAbsent` to avoid the penalty of `computeIfAbsent`.